hoomd/md/CustomForceCompute.h

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: joaander
+
+#include "hoomd/ForceCompute.h"
+
+#include <map>
+#include <memory>
+
+/*! \file CustomForceCompute.h
+    \brief Declares the backend for computing custom forces in python classes
+*/
+
+#ifdef __HIPCC__
+#error This header cannot be compiled by nvcc
+#endif
+
+#include <pybind11/pybind11.h>
+
+#ifndef __CUSTOMFORCECOMPUTE_H__
+#define __CUSTOMFORCECOMPUTE_H__
+
+namespace hoomd
+    {
+namespace md
+    {
+//! Adds a custom force
+/*! \ingroup computes
+ */
+class PYBIND11_EXPORT CustomForceCompute : public ForceCompute
+    {
+    public:
+    //! Constructs the compute
+    CustomForceCompute(std::shared_ptr<hoomd::SystemDefinition> sysdef);
+
+    //! Destructor
+    ~CustomForceCompute();
+
+    //! Set the python callback
+    void setCallback(pybind11::object py_callback)
+        {
+        m_callback = py_callback;
+        }
+
+    const GlobalArray<Scalar4>& getForceArray() const
+        {
+        return m_force;
+        }
+
+    const GlobalArray<Scalar>& getVirialArray() const
+        {
+        return m_virial;
+        }
+
+    const GlobalArray<Scalar4>& getTorqueArray() const
+        {
+        return m_torque;
+        }
```

Comment:
These are defined in the base class. Use those instead of duplicating code.

Reply:
Needed to write a version of the getters which are declared `const` and return `const &` in order to make the templating for the `LocalForceComputeData` class to work. I'll try moving the `LocalForceComputeData` to be templated on the `ForceCompute` class before deciding how to address this.

Reply:
Can we just have one method in the base class that is `const`? It should have been that way from the beginning.

Reply:
Can do